go-sdk/pkg/worker/task.go

```diff
@@ -0,0 +1,158 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package worker
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"reflect"
+	"runtime"
+
+	"github.com/apache/airflow/go-sdk/sdk"
+)
+
+type (
+	taskFunction struct {
+		fn       reflect.Value
+		fullName string
+	}
+	Task interface {
+		Execute(ctx context.Context, logger *slog.Logger) error
+	}
+)
+
+var _ Task = (*taskFunction)(nil)
+
+func NewTaskFunction(fn any) (Task, error) {
+	v := reflect.ValueOf(fn)
+	fullName := runtime.FuncForPC(v.Pointer()).Name()
+	f := &taskFunction{v, fullName}
+	return f, f.validateFn(v.Type())
+}
+
+func (f *taskFunction) Execute(ctx context.Context, logger *slog.Logger) error {
+	fnType := f.fn.Type()
+
+	reflectArgs := make([]reflect.Value, fnType.NumIn())
+	for i := range reflectArgs {
```

Comment:
Not right now, but worth considering. We do reflection args validation during "execution" here, could we maybe do it during registration? (execution is a hot path)